src/ray/gcs/gcs_server/gcs_task_manager.h

```diff
@@ -168,12 +237,16 @@ class GcsTaskManager : public rpc::TaskInfoHandler {
     absl::flat_hash_map<TaskAttempt, size_t> task_attempt_index_;
 
     /// Secondary index from task id to task attempts.
+    /// TODO(rickyx): use unordered_multimap for these as well.
     absl::flat_hash_map<TaskID, absl::flat_hash_set<TaskAttempt>>
         task_to_task_attempt_index_;
     /// Secondary index from job id to task attempts of the job.
     absl::flat_hash_map<JobID, absl::flat_hash_set<TaskAttempt>>
         job_to_task_attempt_index_;
 
+    /// Secondary index from parent task id to a set of children task ids.
+    std::unordered_multimap<TaskID, TaskID> parent_to_children_task_index_;
```

Comment:
Use absl::btree_multimap for performance. The stdlib variants are often much slower.

Reply:
Ah I see - this adds an ordering constrain on the TaskID if I want to use that directly. But I guess I could store the values as serialzied strings.